### ISSUE TITLE
iris without KNN

### ISSUE THREAD

**Issue by parsaanooshfar-lab** [parsaanooshfar-lab/homework-machine-learning#10]
توضیحات:
با اجرای کد پایین روی دیتاست ایریس مشاهده شد که بیشتر مدل‌ها دقت بسیار بالایی دارند و معمولاً بین ۹۷ تا ۱۰۰ درصد است. مدل‌هایی مانند رگرسیون لجستیک، درخت تصمیم، جنگل تصادفی و ماشین بردار پشتیبان تقریباً همیشه نزدیک به ۱۰۰ درصد دقت دارند، زیرا ویژگی‌های این دیتاست به خوبی قابل تفکیک هستند. مدل بیز ساده کمی کمتر عمل می‌کند، معمولاً حدود ۹۳ تا ۹۷ درصد، زیرا فرض استقلال ویژگی‌ها را دارد و در واقع برخی ویژگی‌های ایریس با هم همبستگی دارند.

به طور کلی، همه مدل‌ها عملکرد خوبی روی این دیتاست کوچک دارند و تفاوت اصلی آن‌ها بیشتر مربوط به نحوه یادگیری و فرضیات هر مدل است تا دقت نهایی.

<img width="1580" height="980" alt="Image" src="https://github.com/user-attachments/assets/5ddbfe23-dc14-459d-bf0d-8f873d99041a" />

# Import کتابخانه‌ها
import matplotlib.pyplot as plt
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import accuracy_score
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.svm import SVC
from sklearn.naive_bayes import GaussianNB

# بارگذاری دیتاست Iris
iris = load_iris()
X = iris.data
y = iris.target

# تقسیم داده‌ها به train و test
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

# استانداردسازی داده‌ها (برای Logistic Regression و SVM ضروری است)
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

# تعریف مدل‌ها
models = {
    "Logistic Regression": LogisticRegression(),
    "Decision Tree": DecisionTreeClassifier(),
    "Random Forest": RandomForestClassifier(),
    "SVM": SVC(),
    "Naive Bayes": GaussianNB()
}

# آموزش و ارزیابی مدل‌ها
results = {}
for name, model in models.items():
    if name in ["Logistic Regression", "SVM"]:
        model.fit(X_train_scaled, y_train)  # استفاده از داده‌های استاندارد
        y_pred = model.predict(X_test_scaled)
    else:
        model.fit(X_train, y_train)
        y_pred = model.predict(X_test)
    acc = accuracy_score(y_test, y_pred)
    results[name] = acc

# نمایش نتایج در کنسول
print("=== مقایسه Accuracy مدل‌ها روی دیتاست Iris ===")
for model_name, accuracy in results.items():
    print(f"{model_name}: Accuracy = {accuracy:.2f}")

# رسم نمودار مقایسه‌ای
plt.figure(figsize=(8,5))
plt.bar(results.keys(), results.values(), color=['skyblue', 'lightgreen', 'salmon', 'violet', 'orange'])
plt.ylim(0, 1)
plt.ylabel("Accuracy")
plt.title("مقایسه Accuracy مدل‌های مختلف روی دیتاست Iris")
for i, v in enumerate(results.values()):
    plt.text(i, v + 0.02, f"{v:.2f}", ha='center', fontweight='bold')
plt.show()
